llvm-bazel/llvm-project-overlay/llvm/BUILD

```diff
@@ -30,2213 +14,1351 @@ exports_files(["LICENSE.TXT"])
+          "echo -e '#undef HANDLE_EXTENSION' >> $@\n",
 )
 
 cc_library(
-    name = "AVRCodeGen",
+    name = "Support",
     srcs = glob([
-        "lib/Target/AVR/*.c",
-        "lib/Target/AVR/*.cpp",
-        "lib/Target/AVR/*.inc",
+        "lib/Support/*.c",
+        "lib/Support/*.cpp",
+        "lib/Support/*.h",
+        "lib/Support/*.inc",
+        "lib/Support/Unix/*.h",
+        "lib/Support/Unix/*.inc",
     ]),
     hdrs = glob([
-        "include/llvm/Target/AVR/*.h",
-        "include/llvm/Target/AVR/*.def",
-        "include/llvm/Target/AVR/*.inc",
-        "lib/Target/AVR/*.h",
-    ]),
-    copts = llvm_copts + ["-Iexternal/llvm-project/llvm/lib/Target/AVR"],
-    deps = [
-        ":AVRDesc",
-        ":AVRInfo",
-        ":AsmPrinter",
-        ":CodeGen",
-        ":Core",
-        ":MC",
-        ":SelectionDAG",
-        ":Support",
-        ":Target",
-        ":config",
+        "include/llvm/Support/*.h",
+        "include/llvm/ADT/*.h",
+    ]) + [
+        "include/llvm/Support/VCSRevision.h",
+        "include/llvm/Support/Extension.def",
+        "include/llvm-c/Core.h",
+        "include/llvm-c/DataTypes.h",
+        "include/llvm-c/DisassemblerTypes.h",
+        "include/llvm-c/Error.h",
+        "include/llvm-c/ExternC.h",
+        "include/llvm-c/ErrorHandling.h",
+        "include/llvm-c/Support.h",
+        "include/llvm-c/Types.h",
     ],
-)
-
-cc_library(
-    name = "AVRDesc",
-    srcs = glob([
-        "lib/Target/AVR/MCTargetDesc/*.c",
-        "lib/Target/AVR/MCTargetDesc/*.cpp",
-        "lib/Target/AVR/MCTargetDesc/*.inc",
-    ]),
-    hdrs = glob([
-        "include/llvm/Target/AVR/MCTargetDesc/*.h",
-        "include/llvm/Target/AVR/MCTargetDesc/*.def",
-        "include/llvm/Target/AVR/MCTargetDesc/*.inc",
-        "lib/Target/AVR/MCTargetDesc/*.h",
-    ]),
-    copts = llvm_copts + ["-Iexternal/llvm-project/llvm/lib/Target/AVR"],
-    deps = [
-        ":AVRInfo",
-        ":MC",
-        ":Support",
-        ":config",
+    copts = llvm_copts,
+    includes = ["include"],
+    linkopts = [
+        "-lz",
+        "-ltinfo",
+        "-pthread",
     ],
-)
-
-cc_library(
-    name = "AVRDisassembler",
-    srcs = glob([
-        "lib/Target/AVR/Disassembler/*.c",
-        "lib/Target/AVR/Disassembler/*.cpp",
-        "lib/Target/AVR/Disassembler/*.inc",
-    ]),
-    hdrs = glob([
-        "include/llvm/Target/AVR/Disassembler/*.h",
-        "include/llvm/Target/AVR/Disassembler/*.def",
-        "include/llvm/Target/AVR/Disassembler/*.inc",
-        "lib/Target/AVR/Disassembler/*.h",
+    textual_hdrs = glob([
+        "include/llvm/Support/*.def",
     ]),
-    copts = llvm_copts + ["-Iexternal/llvm-project/llvm/lib/Target/AVR"],
+    visibility = ["//visibility:public"],
     deps = [
-        ":AVRInfo",
-        ":MCDisassembler",
-        ":Support",
+        ":Demangle",
         ":config",
     ],
 )
 
 cc_library(
-    name = "AVRInfo",
+    name = "LineEditor",
```

Comment:
We could probably rig something up with repo rules. I think I'd have to understand better what we're testing and what we're guarding against. I'm not too worried, I was just surprised that this seemed to escape the sandbox.

Reply:
Ah we actually run into the same issue with zlib. The build now fails on IREE's remote executors building `Support` complaining about zlib.h I assume because they don't have zlib.

Can you add the "@zlib" dependency back to `Support`? Or should this just be an installed dependency? https://reviews.llvm.org/D79219 also has some discussion about detecting zlib.

Reply:
Ok I think I don't understand well enough what these things are being used for to understand why it *needs* to be a system lib, but we can discuss later :-)